lib/iris/tests/test_pandas.py

```diff
@@ -491,13 +502,427 @@ def test_copy_true(self):
+        (result_cube,) = result
+        result_coord = result_cube.coord(coord_name)
+        self.assertArrayEqual(result_coord.points, expected_points)
+        assert result_coord.cube_dims(result_cube) == (0, 1)
+
+    def test_coord_varies_all_indices(self):
+        df = self._create_pandas(index_levels=3)
+        coord_shape = df.index.levshape
+        coord_values = np.arange(np.product(coord_shape))
+        coord_name = "foo"
+        df[coord_name] = coord_values
+        result = iris.pandas.as_cubes(df, aux_coord_cols=[coord_name])
+
+        expected_points = coord_values.reshape(coord_shape)
+        (result_cube,) = result
+        result_coord = result_cube.coord(coord_name)
+        self.assertArrayEqual(result_coord.points, expected_points)
+        assert result_coord.cube_dims(result_cube) == (0, 1, 2)
+
+    def test_category_coord(self):
+        # Something that varies on a dimension, but doesn't change with every
+        #  increment.
+        df = self._create_pandas(index_levels=2)
+        coord_shape = df.index.levshape
+        coord_values = np.arange(np.product(coord_shape))
+        coord_name = "foo"
+
+        # Create a repeating value along a dimension.
+        step = coord_shape[-1]
+        coord_values[1::step] = coord_values[::step]
+
+        df[coord_name] = coord_values
+        result = iris.pandas.as_cubes(df, aux_coord_cols=[coord_name])
+
+        expected_points = coord_values.reshape(coord_shape)
+        (result_cube,) = result
+        result_coord = result_cube.coord(coord_name)
+        self.assertArrayEqual(result_coord.points, expected_points)
+        assert result_coord.cube_dims(result_cube) == (0, 1)
+
+    def test_scalar_coord(self):
+        df = self._create_pandas()
+        coord_values = np.ones(len(df))
+        coord_name = "foo"
+        df[coord_name] = coord_values
+        result = iris.pandas.as_cubes(df, aux_coord_cols=[coord_name])
+
+        expected_points = np.unique(coord_values)
+        (result_cube,) = result
+        result_coord = result_cube.coord(coord_name)
+        self.assertArrayEqual(result_coord.points, expected_points)
+        assert result_coord.cube_dims(result_cube) == tuple()
+
+    def test_multi_phenom(self):
+        df = self._create_pandas()
+        new_name = "new_phenom"
+        df[new_name] = df[0]
+        result = iris.pandas.as_cubes(df)
+
+        # Note the shared coord object between both Cubes.
+        expected_coord = DimCoord(df.index.values)
+        expected_cube_kwargs = dict(dim_coords_and_dims=[(expected_coord, 0)])
+
+        expected_cube_0 = Cube(
+            data=df[0].values,
+            long_name=str(df[0].name),
+            **expected_cube_kwargs,
+        )
+        expected_cube_1 = Cube(
+            data=df[new_name].values,
+            long_name=new_name,
+            **expected_cube_kwargs,
+        )
+        assert result == [expected_cube_0, expected_cube_1]
+
+    def test_no_phenom(self):
```

Comment:
Should we also care about an empty dataframe. Currently we get:
```
>>> my_df = pd.DataFrame({})
>>> result = iris.pandas.as_cubes(my_df)
Traceback (most recent call last):
  File ".../code.py", line 90, in runcode
    exec(code, self.locals)
  File "<input>", line 1, in <module>
  File "/net/data/users/ldreyer/dev/iris/lib/iris/pandas.py", line 419, in as_cubes
  File ".../iris/lib/iris/pandas.py", line 407, in format_dimensional_metadata
  File ".../iris/lib/iris/pandas.py", line 62, in _get_dimensional_metadata
    coord = AuxCoord(points, units=units)
  File ".../site-packages/cf_units/__init__.py", line 1970, in date2num
    return cftime.date2num(date, self.cftime_unit, self.calendar)
  File "src/cftime/_cftime.pyx", line 287, in cftime._cftime.date2num
IndexError: list index out of range
````

Reply:
Oooh good catch! Do you think we should:

a. Throw an `Error`?
b. Return an empty `CubeList`?

Reply:
We went for an empty `CubeList` - 9619fd065f381c89e58f1383a2a2ccbced3adfdc

Is the pattern of using an early `return` OK? I know we typically `return result` having conditionally assigned a value to `result`, but I didn't like the idea of almost everything being within an `if` block.